code/game/atoms/atom-materials.dm

```diff
@@ -44,6 +44,8 @@
 /// this is only to be used if the material was registered. if it was never registered, DO NOT call this.
 /// this should null-check, as null is a valid material.
 /atom/proc/unregister_material(datum/prototype/material/mat, primary)
+	if(!istype(mat)) //null checking
```

Comment:
below already null-checks 
also do !mat

passing in non-null but non-mat is an error state and should runtime